client/client-lib/src/mint/backup.rs

```diff
@@ -321,6 +326,49 @@ impl MintClient {
     }
 }
 
+/// Backup metadata
+///
+/// A backup can have a blob of extra data encoded in it.
+/// We provide methods to use json encoding, but in clients
+/// are free to use their own encoding.
+#[derive(Serialize, Deserialize, PartialEq, Eq, Debug, Encodable, Decodable, Clone)]
+pub struct Metadata(Vec<u8>);
+
+impl Metadata {
+    /// Create empty metadata
+    pub fn empty() -> Self {
+        Self(vec![])
+    }
+
+    pub fn from_raw(bytes: Vec<u8>) -> Self {
+        Self(bytes)
+    }
+
+    pub fn into_raw(self) -> Vec<u8> {
+        self.0
+    }
+
+    /// Is metadata empty
+    pub fn is_empty(&self) -> bool {
+        self.0.is_empty()
+    }
+
+    /// Create metadata as json from typed `val`
+    pub fn json_from<T: Serialize>(val: T) -> Self {
```

Comment:
isn't that usually called `from_json`?

Reply:
It's doing something else semantically: taking an object, encoding it as JSON and putting the resulting bytes into metadata. I might call it `from_json_encoded`.